### ISSUE TITLE
Layer Accessibility 

### ISSUE THREAD

**Issue by britt6612** [grommet/grommet#5976]
### Expected Behavior
When opening a `layer` thinking there should of some sort of announcement that a user using a screen reader would know that they are in a `layer` and how to return out of the `layer`

when using a screen reader and opening a `layer` wondering if we can add something such as a a11ytitle to let the user know they are now in a layer
### Actual Behavior
you have to manually put an a11ytitle on a close button for example in the layer and autofocus on that button for the user to know 

### URL, screen shot, or Codepen exhibiting the issue

https://design-system.hpe.design/components/layer
First example we have 
```
                a11yTitle={`You are on a Close button in a layer containing
                a text description. To close the layer 
                and return to the primary content, press Enter.`}
                autoFocus
```
on the close `button`

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version: latest
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by eatyourgreens** [comment]
When a popup opens in a Layer, the dialog role seems like the most appropriate role to use. A screen reader should announce that a dialog has opened. I’ve been hacking a component that uses Grommet Layer to make sure that it has the dialog role (and an accessible name.) I’ve also found that I have to manually manage pass focus to a Layer when it opens. 
- https://github.com/zooniverse/front-end-monorepo/pull/6796

**Comment by jcfilben** [comment]
We should still explore potentially adding `role='dialog'` in #6855. I think we can close this issue because it feels reasonable to expect that the implementer needs to pass an aria-label on the close button
